CodeGen/Generators/QuantityJsonFilesParser.cs

```diff
@@ -124,5 +126,246 @@ private static Localization[] GetLocalizationForPrefixUnit(IEnumerable<Localizat
                 };
             }).ToArray();
         }
+
+        private static BaseUnits? GetPrefixedBaseUnits(BaseDimensions dimensions, BaseUnits? oldUnits, PrefixInfo prefixInfo)
+        {
+            if (oldUnits is null)
+            {
+                return null;
+            }
+
+            // iterate the non-zero dimensions in the order [1, -1, 2, -2...n, -n]
+            foreach (var degree in dimensions.GetNonZeroDegrees().OrderBy(int.Abs).ThenByDescending(x => x))
+            {
+                if (TryPrefixWithDegree(dimensions, oldUnits, prefixInfo.Prefix, degree, out BaseUnits? prefixedUnits))
+                {
+                    return prefixedUnits;
+                }
+            }
+            
+            return null; 
+        }
```

Comment:
Note that when we have multiple dimensions present, there are often multiple combinations of base units that can fit a given prefix scale. This algorithm tries the smallest dimensions first, with a preference for the positive exponents.

Technically, if no exact scaling factor exists for a given quantity, having more than one dimensions- there could be another iteration after the `foreach` loop that tries to simultaneously change _two dimensions_ with a single prefix - but that would have really made my head hurt..